File: pset6/credit.py
#!/usr/bin/env python3

import cs50

def main():
    while(True):
        print("Card number: ",end="")
        ccnumber = cs50.get_int()
        if ccnumber>0:
            break
       
    multiply = False
    fnumber = 0
    count=0
    while ccnumber!=0:
        digit = ccnumber%10
        ccnumber //= 10
    
        if multiply==True:
            digit*=2
            fnumber+=digit%10
            fnumber+=digit//10
            multiply=False
        else:
            fnumber+=digit
            multiply=True
        count+=1
        if ccnumber<100 and ccnumber>10:
            first_two = ccnumber

    if fnumber%10==0 and count>12 and count<17:
        if first_two==37 or first_two==34:
            print("AMEX")
        
        elif first_two>50 and first_two<56:
            print("MASTERCARD")
        
        elif first_two>=40 and first_two<50:
            print("VISA")
    else:
            print("INVALID")    
    
if __name__ == "__main__":
    main()

File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>


int main(void)
{
    long long cc_number;
    do{
    printf("Number: ");
    cc_number = get_long_long();
    }while(cc_number<0);
    
    int digit;
    bool multiple = false;
    int final_number = 0;
    int first_two = 0;
    int count=0;
    
    while(cc_number != 0)
    {
        digit =  cc_number %10;
        cc_number /= 10;
        
        if(multiple){
            digit *= 2;
            final_number += digit%10;
            final_number += digit/10;
        }
        else{
        final_number += digit;
        }
        multiple = (multiple) ? false : true ;
        if(cc_number<100 & cc_number>10){
            first_two = cc_number;
        }
        ++count;
        }
    if(final_number%10==0 & count>12 & count<17){
        if(first_two==37 || first_two==34){
            printf("AMEX\n");
        }
        else if(first_two>50 & first_two<56){
            printf("MASTERCARD\n");
        }
        else if(first_two>=40 & first_two<50){
            printf("VISA\n");
        }
    }
    else{
            printf("INVALID\n");    
    }
    
}